systems/controllers/QPControllermex.cpp

```diff
@@ -304,23 +302,27 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
   
   pdata->r->getCOM(xcom);
   pdata->r->getCOMJac(pdata->J);
-  pdata->r->getCOMJacDot(pdata->Jdot);
+  MatrixXd Jdot;
+  pdata->r->getCOMJacDot(Jdot);
+  pdata->Jdotv = Jdot*qd;
   pdata->J_xy = pdata->J.topRows(2);
-  pdata->Jdot_xy = pdata->Jdot.topRows(2);
+  pdata->Jdotv_xy = pdata->Jdotv.head(2);

```

Comment:
for these, using the fixed size version could be alittle faster (e.g. `Jdotv.head<2>()`)